osidb/serializer.py

```diff
@@ -861,6 +862,69 @@ class Meta:
             + TrackingMixinSerializer.Meta.fields
         )
 
+    def update(self, affect, validated_data):
+        """
+        perform the tracker instance update
+        """
+
+        # NOTE: This overrides update() behavior provided by mixins, but mostly
+        #       mirrors the mixins. If mixins change, this method must be updated.
+        # NOTE: Look at TrackerSerializer.update() and
+        #       FlawPackageVersionSerializerMixin.update() for similar ways of
+        #       interposing actions within the mixins' update() MRO & flow.
+
+        ############################
+        # 1) prepare prerequisites #
+        ############################
+
+        # From ACLMixinSerializer.update()
+        validated_data = self.embargoed2acls(validated_data)
+
+        #########################
+        # 2) pre-update actions #
+        #########################
+
+        old_affectedness = affect.affectedness
+
+        #####################
+        # 3) update actions #
+        #####################
+
+        # From BugzillaSyncMixinSerializer.update()
+        for attr, value in validated_data.items():
+            setattr(affect, attr, value)
+
+        # From BugzillaBareSyncMixinSerializer.update()
+        affect.save(bz_api_key=self.get_bz_api_key())
+
+        ##########################
+        # 4) post-update actions #
+        ##########################
+
+        # When an affect goes from NEW affectedness to something else, any related tracker
+        # must have its validation-requested label removed, if it had one.
+        # Since the label is not stored in osidb, removal is done by regenerating the label
+        # by re-saving the tracker.
+
+        if (
+            old_affectedness != affect.affectedness
+            and old_affectedness == Affect.AffectAffectedness.NEW
+        ):
```

Comment:
can be changed to `if old_affectedness == Affect.AffectAffectedness.NEW != affect.affectedness:`

Reply:
It can, but I'd rather not. It's technically correct, but its correctness seems subjectively so unintuitive seeing such an expression stops me for some time thinking about it. Python is AFAIK an outlier language in [doing that correctly](https://docs.python.org/3/reference/expressions.html#grammar-token-python-grammar-or_expr:~:text=Formally%2C%20if%20a%2C%20b%2C%20c%2C%20%E2%80%A6%2C%20y%2C%20z%20are%20expressions%20and%20op1%2C%20op2%2C%20%E2%80%A6%2C%20opN%20are%20comparison%20operators%2C%20then%20a%20op1%20b%20op2%20c%20...%20y%20opN%20z%20is%20equivalent%20to%20a%20op1%20b%20and%20b%20op2%20c%20and%20...%20y%20opN%20z). In e.g. C that would be illegal and in Javascript, it is a left-to-right series of individual comparisons that results in expressions like `3 == 3 == 1` and `3 === 3 === true` being true.

Reply:
Chained comparisons are pretty common in Python, and you can read this one directly the way it is written:
Do something if `old affectedness is NEW and this is different from the current affectedness` (i.e. `A == B and B =! C` is simplified to `A == B != C`).

I think that the argument regarding invalidity in other languages is not fair. For example, somebody can use increment operator ++ in Java even though this is invalid in Python. But again, I will leave it up to you.

Reply:
Changed based on your feedback.